extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -81,13 +105,13 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
    */
   createAndAppendInnerShoppingTagEl_(shoppingData) {
     const tagData = shoppingData[this.element.getAttribute('data-tag-id')];
-    if (!tagData) {
+    if (this.renderedEl_ || !tagData) {
       return;
     }
     this.mutateElement(() => {
-      createShadowRootWithStyle(
+      this.renderedEl_ = createShadowRootWithStyle(
```

Comment:
`createShadowRootWithStyle()` actually returns the first parameter instead of the second parameter, so `renderedEl_` is being set to the `this.element` now. So, you would need to update `renderedEl_` to point to the correct element.

However, since `this.renderedEl_` is only really being used in this class to determine if the inner shopping tag has already been appended, consider replacing it with a boolean property like `this.hasAppendedInnerShoppingTagEl_` that is initialized to `false` and then set to `true` right after the `createShadowRootWithStyle()` call.

Reply:
Sounds good, I will refactor the code to reflect these changes.